src/app/loadout-builder/item-filter.ts

```diff
@@ -94,14 +97,45 @@ export function filterItems(
 function matchedLockedModEnergy(
   defs: D2ManifestDefinitions,
   item: DimItem,
-  lockedModsByPlugCategoryHash: PluggableInventoryItemDefinition[],
+  lockedMods: PluggableInventoryItemDefinition[] | undefined,
   upgradeSpendTier: UpgradeSpendTier,
   lockItemEnergyType: boolean
 ) {
-  if (!lockedModsByPlugCategoryHash) {
+  if (!lockedMods) {
     return true;
   }
-  return lockedModsByPlugCategoryHash.every((mod) =>
+  return lockedMods.every((mod) =>
     doEnergiesMatch(defs, mod, item, upgradeSpendTier, lockItemEnergyType)
   );
 }
+
+function hasEnoughSocketsForMods(
```

Comment:
Not 100% clear on what this is doing

Reply:
I will write some docs for it, this was put together pretty quick to get early feedback and try get it ready for release. Basically its just ensuring each item has enough sockets for the mods selected.

We need to ditch the `plugCateogryHash` method of identifying mod/item compatibility. The artificer mods have the same `plugCategoryHash` as the bucket specific mod sockets but they have a subset of mods that can socket into them. The only way to check that now is to pull the plug sets and ensure the mod hash is present in the plug set.